packages/opencensus-instrumentation-http/src/http.ts

```diff
@@ -360,12 +360,28 @@ export class HttpPlugin extends BasePlugin {
 
       const setter: HeaderSetter = {
         setHeader(name: string, value: string) {
-          request.setHeader(name, value);
+          // https://github.com/googleapis/cloud-trace-nodejs/pull/766
+          if (plugin.hasExpectHeader(options) && options.headers) {
+            options.headers[name] = value;
+          } else {
+            request.setHeader(name, value);
+          }
         }
       };
 
       const propagation = plugin.tracer.propagation;
       if (propagation) {
+        // If outgoing request headers contain the "Expect" header, the returned
+        // ClientRequest will throw an error if any new headers are added. For
+        // this reason, only in this scenario, we opt to clone the options
+        // object to inject the trace context header instead of using
+        // ClientRequest#setHeader. (We don't do this generally because cloning
+        // the options object is an expensive operation.)
+        // https://github.com/googleapis/cloud-trace-nodejs/pull/766
```

Comment:
The explanation above is great - can we remove the PR reference? (People can use the blame layer to find it if needed)